aws/signer/v4/v4.go

```diff
@@ -311,6 +311,42 @@ func (v4 Signer) Presign(r *http.Request, body io.ReadSeeker, service, region st
 	return v4.signWithBody(r, body, service, region, exp, true, signTime)
 }
 
+// SignString creates a signature of provided string according to AWS v4
+// signing process described in the documentation:
+// https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html
+//
+// It may be used to create signatures of "Browser-Based Upload using HTTP POST"
+// policy documents to be used in the browser. In which case, make sure to
+// base64 encode the policy document using base64.StdEncoding before creating
+// it's signature.
```

Comment:
Might be worth linking to `https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-authentication-HTTPPOST.html` as well in this comment that describe the browser-based uploads use-case.